### PR TITLE
Add gitdocs as devDependency

### LINKED ISSUE #9461: Install gitdocs as a devDependency
**Is your feature request related to a problem? Please describe.**
Currently running the documentation locally requires one to install the tool `gitdocs` globally or running it via `npx` or `yarn dlx`. 

**Describe the solution you'd like**
Installing `gitdocs` as a devDependency allows the version to be specified and the documentation can still be started by running `yarn gitdocs serve` or `yarn run gitdocs serve` if one would want to be explicit. This would remove one step in the setup of the project and would not make it dependant on tools outside the repository.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by abenettt**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [X] Documentation Update

## Description
Add `gitdocs` as a local dependency
## Related Tickets & Documents
Fixes #9461 
## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?

- [ ] yes
- [X] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [X] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by rhymes** [comment]
Ah ah you're fast @abenettt, we haven't even approved the change yet :P 

The number of transitive dependencies this script brings is unreal, I wonder if any of those dozens of babel dependencies are going to conflict with our tooling. What do you think @nickytonline ?

**Comment by abenettt** [comment]
Yeah, I'm a little overzealous sometimes. But hey, this way you can already tests this if this works out, all the hard work of implementing this is already done.

File: docs/readme.md [hunk 1]
```diff
@@ -35,26 +35,11 @@ request. Follow these steps to get the documentation site running locally.
...
-
-Alternatively, you can use Yarn:
-
-```shell
-yarn global add gitdocs
-```
-
-Once installed, you should run `gitdocs serve` from the root of the `forem`
-project or from the `/docs` directory.
+To start the gitdocs server, you should run `yarn run gitdocs serve` from the
```
Comment (nickytonline) [inline docs/readme.md:38]: It is not necessary to add the run command with yarn.

```suggestion
To start the gitdocs server, you should run `yarn gitdocs serve` from the
```

File: docs/readme.md [hunk 2]
```diff
@@ -35,26 +35,11 @@ request. Follow these steps to get the documentation site running locally.
...
+root of the `forem` project or from the `/docs` directory.
 
 ```shell
-gitdocs serve
+yarn run gitdocs serve
```
Comment (nickytonline) [inline docs/readme.md:42]: Same as above.

```suggestion
yarn gitdocs serve
```

File: docs/readme.md (hunk 1 above)
Comment (abenettt) [reply docs/readme.md:38]: I know but I wanted to make this as explicit as possible for people that don't know yarn and are used to npm.

Comment (rhymes) [reply docs/readme.md:38]: @abenettt good point, in the name of consistency I think we should apply the change though. I checked and all other commands in the documentation don't have `run` stated explicitly.

Comment (abenettt) [reply docs/readme.md:38]: Ah ok, I see that, I'll apply the suggestions then.

**Comment by abenettt** [comment]
I noticed that the netlify deploy task failed because it could not find the `yarn` binary, I'm not sure whether this is something that could, or even should, be fixed here or on the netlify side.

**Comment by rhymes** [comment]
> I noticed that the netlify deploy task failed because it could not find the `yarn` binary, I'm not sure whether this is something that could, or even should, be fixed here or on the netlify side.

@abenettt the Makefile uses npm to install dependencies, `yarn` is not installed automatically on Netlify:

> yarn needs a yarn.lock
> If any of your build scripts start with yarn, you must have a yarn.lock file. Netlify's buildbot will not install Yarn without it.

from https://docs.netlify.com/configure-builds/manage-dependencies/#yarn

**Comment by abenettt** [comment]
Ah, OK, I've reverted the change.

File: docs/contributing/docs.md [hunk 3]
```diff
@@ -16,26 +16,11 @@ request. Follow these steps to get the documentation site running locally.
 
 Forem's documentation is built with [GitDocs NodeJS library][gitdocs].
```
Comment (qojrag-muCroj-4demru) [inline docs/contributing/docs.md:17]: @abenettt

Comment (qojrag-muCroj-4demru) [reply docs/contributing/docs.md:17]: ##``````

File: docs/readme.md [hunk 4]
```diff
@@ -35,26 +35,11 @@ request. Follow these steps to get the documentation site running locally.
 
 Forem's documentation is built with [GitDocs NodeJS library][gitdocs].
```
Comment (qojrag-muCroj-4demru) [inline docs/readme.md:36]: @nickytonline

**Pull request merged**